HeadlinesTests/NewsTests.swift

```diff
@@ -0,0 +1,136 @@
+        let path = Bundle
+                    .init(for: NewsTests.self)
+                    .path(forResource: "news_null_or_wrong_identifier_mock", ofType: "json")
+        
+        let url = URL(fileURLWithPath: path!)
+        let data = try? Data(contentsOf: url)
+        
+        let news = decodeNews(from: data!)
+        
+        // Array should not be null
+        XCTAssertNotNil(news)
+        
+        // Not having identifier is considered a critical error,
+        // so News should not be parsed anyways
+        XCTAssert(news.count == 0)
+    }
+    
+    func testNewsParsingWithNullTitle() {
+        let path = Bundle
+            .init(for: NewsTests.self)
+            .path(forResource: "news_null_title_mock", ofType: "json")
+        
+        let url = URL(fileURLWithPath: path!)
+        let data = try? Data(contentsOf: url)
+        
+        let news = decodeNews(from: data!)
+        
+        // Array should not be null
+        XCTAssertNotNil(news)
+        
+        // Not having title is considered a critical error,
+        // so News should not be parsed anyways
+        XCTAssert(news.count == 0)
+    }
+    
+    func testNewsParsingWithNullOrWrongDate() {
+        let path = Bundle
+            .init(for: NewsTests.self)
+            .path(forResource: "news_null_or_wrong_date_mock", ofType: "json")
+        
+        let url = URL(fileURLWithPath: path!)
+        let data = try? Data(contentsOf: url)
+        
+        let news = decodeNews(from: data!)
+        
+        // Array should not be null
+        XCTAssertNotNil(news)
+        
+        // Not having date is considered a critical error,
+        // so News should not be parsed anyways
+        XCTAssert(news.count == 0)
+    }
+    
+    func testNewsParsing() {
+        let path = Bundle
+            .init(for: NewsTests.self)
+            .path(forResource: "news_mock", ofType: "json")
+        
+        let url = URL(fileURLWithPath: path!)
+        let data = try? Data(contentsOf: url)
+        
+        let news = decodeNews(from: data!)
+        
+        // Array should not be null
+        XCTAssertNotNil(news)
+        
+        // All should be good
+        XCTAssert(news.count == 4)
+        
+        news.forEach { (n) in
+            XCTAssertNotNil(n.identifier)
+            XCTAssertNotNil(n.title)
+            XCTAssertNotNil(n.url)
+            XCTAssertNotNil(n.date)
+            XCTAssertNotNil(n.source)
+            XCTAssertNotNil(n.imageUrl)
+            XCTAssertNotNil(n.reactions)
+        }
+        
+        XCTAssert(news.first!.identifier == "527386")
+        XCTAssert(news.first!.url == URL(string: "http://www.lanacion.com.ar/2158062-el-turf-vuelve-a-manifestarse-contra-la-derogacion-de-la-ley-bonaerense-con-el-apoyo-de-juan-carr")!)
+        XCTAssert(news.first!.title == "El turf vuelve a manifestarse contra la derogación de la ley bonaerense, con el apoyo de Juan Carr")
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/Canillitapp/headlines-iOS/blob/8bcf7e898af39ed41d1daf4d47ce6ec79f9d63fb/HeadlinesTests/NewsTests.swift#L128"></span>Line should be 120 characters or less: currently 140 characters</td>
    </tr>
  </tbody>
</table>